rust-src/concordium_base/src/web3id/v1.rs

```diff
@@ -0,0 +1,2092 @@
+/// order to verify the credential.
+// for some reason, version 1.82 clippy thinks the `Identity` variant is 0 bytes and hence gives this warning
+#[allow(clippy::large_enum_variant)]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum CredentialV1<
+    P: Pairing,
+    C: Curve<Scalar = P::ScalarField>,
+    AttributeType: Attribute<C::Scalar>,
+> {
+    /// Credential based on an on-chain account
+    Account(AccountBasedCredentialV1<C, AttributeType>),
+    /// Identity based credential
+    Identity(IdentityBasedCredentialV1<P, C, AttributeType>),
+}
+
+impl<
+        P: Pairing,
+        C: Curve<Scalar = P::ScalarField>,
+        AttributeType: Attribute<C::Scalar> + serde::Serialize,
+    > serde::Serialize for CredentialV1<P, C, AttributeType>
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            Self::Account(AccountBasedCredentialV1 {
+                subject,
+                issuer,
+                proof: proofs,
+            }) => {
+                let mut map = serializer.serialize_map(None)?;
+                map.serialize_entry(
+                    "type",
+                    &[
+                        VERIFIABLE_CREDENTIAL_TYPE,
+                        CONCORDIUM_VERIFIABLE_CREDENTIAL_V1_TYPE,
+                        CONCORDIUM_ACCOUNT_BASED_CREDENTIAL_TYPE,
+                    ],
+                )?;
+                map.serialize_entry("credentialSubject", subject)?;
+                let issuer = did::Method::new_idp(subject.network, *issuer);
+                map.serialize_entry("issuer", &issuer)?;
+                map.serialize_entry("proof", proofs)?;
+                map.end()
+            }
+            Self::Identity(IdentityBasedCredentialV1 {
+                issuer,
+                validity,
+                subject,
+                proof: proofs,
+            }) => {
+                let mut map = serializer.serialize_map(None)?;
+                map.serialize_entry(
+                    "type",
+                    &[
+                        VERIFIABLE_CREDENTIAL_TYPE,
+                        CONCORDIUM_VERIFIABLE_CREDENTIAL_V1_TYPE,
+                        CONCORDIUM_IDENTITY_BASED_CREDENTIAL_TYPE,
+                    ],
+                )?;
+                map.serialize_entry("credentialSubject", subject)?;
+                map.serialize_entry(
+                    "validFrom",
+                    &validity
+                        .created_at
+                        .lower()
+                        .ok_or(S::Error::custom("convert crated at to date time"))?,
+                )?;
+                map.serialize_entry(
+                    "validUntil",
+                    &validity
+                        .valid_to
+                        .upper_inclusive()
+                        .ok_or(S::Error::custom("convert valid to to date time"))?,
+                )?;
```

Comment:
Most of the time the `created_at` timestamp of a credential might be equivalent to the `valid_from` timestamp (when the document becomes active/valid) but could be that some credentials/documents are issued where both timestamp are not equal (not sure how all governments in the world handle these timestamps). 

But could be also that our `created_at` is field was just a bad name and it actually means `valid_from`. So not feeling strongly about it.